src/modules/fundingManager/depositVault/FM_DepositVault_v1.sol

```diff
@@ -0,0 +1,130 @@
+
+// External Dependencies
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   Deposit Vault Funding Manager
+ *
+ * @notice  This contract allows users to deposit tokens to fund the workflow.
+ *
+ * @dev     Implements {IFundingManager_v1} interface.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+contract FM_DepositVault_v1 is
+    IFundingManager_v1,
+    IFM_DepositVault_v1,
+    Module_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId == type(IFundingManager_v1).interfaceId
+            || interfaceId == type(IFM_DepositVault_v1).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+
+    /// @dev The token that is deposited.
+    IERC20 private _token;
+
+    /// @dev Storage gap for future upgrades
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Init Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata,
+        bytes memory configData
+    ) external override initializer {
+        __Module_init(orchestrator_, metadata);
+
+        address orchestratorTokenAddress = abi.decode(configData, (address));
+        _token = IERC20(orchestratorTokenAddress);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public View Functions
+
+    /// @inheritdoc IFundingManager_v1
+    function token() public view returns (IERC20) {
+        return _token;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function deposit(uint amount) external {
+        address from = _msgSender();
+        _deposit(from, amount);
+    }
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function depositFor(address from, uint amount) external {
+        _deposit(from, amount);
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @dev Internal function to deposit tokens.
+    /// @param from The address to deposit tokens for.
+    /// @param amount The number of tokens to deposit.
+    function _deposit(address from, uint amount) internal {
```

Comment:
Should probably disallow also 0 amount here

Reply:
Yeah, fair. Not a big fan of wasting gas for every user just to ensure that a potential 0 depositing user doesn't pay gas/send their tx if they forgot to put something in the `amount` field, but I believe we do it in other contracts as well, so I guess that's a fair point.

As a 0-deposit has literally zero impact here though and should just be filtered on the UI/SDK level, I'd personally lean towards not changing anything, unless @Zitzak or @0xNuggan think it makes sense.

Or did I miss any security-related implication you had in mind @ben-kaufman?

Reply:
No that's mostly a matter of prefence, shouldn't matter much

Reply:
Okay, thank you 🙏 I personally wouldn't change anything then, but I'll wait for @Zitzak or @0xNuggan to chime in briefly. 👍